omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -376,10 +376,24 @@ public static void setConcept(Obs obs, Object value) {
 	public static void setValue(Obs obs, Object value) throws ParseException, ConversionException, IOException {
 		if (value != null) {
 			if (obs.isComplex()) {
-				byte[] bytes = DatatypeConverter.parseBase64Binary(value.toString());
+				byte[] bytes = null;
 				
-				ComplexData complexData = new ComplexData(obs.getUuid() + ".raw", new ByteArrayInputStream(bytes));
+				String extension = "raw";
+				//Try to extract extension from the MIME type if it is provided in value
+				if (value.toString().indexOf("data:") == 0) {
```

Comment:
The above test is a bit confusing at the first glance. I thought you are testing for absence of mime type with a return value of 0. It took me time to discover that you are actually checking if it starts with it. Don't you think startsWith would be clearer?